openBCIBoard.js

```diff
@@ -856,180 +1013,174 @@ function OpenBCIFactory() {
-                                    impedanceTestApplyRaw.call(this,rawValue);
-                                }).catch(err => {
-                                    console.log('Impedance calculation error: ' + err);
+
+            // cbuffer
+            var bytesToRead = sizeOfData;
+
+            // is there old data?
+            if (this.buffer) {
+                // Get size of old buffer
+                var oldBufferSize = this.buffer.byteLength;
+
+                // Make a new buffer
+                var newDataBuffer = new Buffer(bytesToRead + oldBufferSize);
+
+                // Put old buffer in the front of the new buffer
+                var oldBytesWritten = this.buffer.copy(newDataBuffer);
+
+                // Move the incoming data into the end of the new buffer
+                data.copy(newDataBuffer,oldBytesWritten);
+
+                // Over write data
+                data = newDataBuffer;
+
+                // Update the number of bytes to read
+                bytesToRead += oldBytesWritten;
+            }
+
+            var readingPosition = 0;
+
+            // 45 < (200 - 33) --> 45 < 167 (good) | 189 < 167 (bad) | 0 < (28 - 33) --> 0 < -5 (bad)
+            while (readingPosition <= bytesToRead - k.OBCIPacketSize) {
+                if (data[readingPosition] === k.OBCIByteStart) {
+                    var rawPacket = data.slice(readingPosition, readingPosition + k.OBCIPacketSize);
+                    if (data[readingPosition + k.OBCIPacketSize - 1] === k.OBCIByteStop) {
+                        // standard packet!
+                        openBCISample.parseRawPacket(rawPacket,this.channelSettingsArray)
+                            .then(sampleObject => {
+
+                                sampleObject._count = this.sampleCount++;
+                                if(this.impedanceTest.active) {
+                                    var impedanceArray;
+                                    if (this.impedanceTest.continuousMode) {
+                                        //console.log('running in contiuous mode...');
+                                        //openBCISample.debugPrettyPrint(sampleObject);
+                                        impedanceArray = openBCISample.goertzelProcessSample(sampleObject,this.goertzelObject)
+                                        if (impedanceArray) {
+                                            this.emit('impedanceArray',impedanceArray);
+                                        }
+                                    } else if (this.impedanceTest.onChannel != 0) {
+                                        // Only calculate impedance for one channel
+                                        impedanceArray = openBCISample.goertzelProcessSample(sampleObject,this.goertzelObject)
+                                        if (impedanceArray) {
+                                            this.impedanceTest.impedanceForChannel = impedanceArray[this.impedanceTest.onChannel - 1];
```

Comment:
Because this is simply updating a value in the array that will be emitted at the end of the single channel test. See `.impedanceTestChannelInputN()` for an example how it resolves the impedance value.